### PR TITLE
Fix patch for go-tuf, replace as per recommendation

### PR DESCRIPTION
**Pull request by vasireddy99**
**Description:** 

This PR will replace the go-tuf dep  as per reference [here](https://github.com/DataDog/datadog-agent/blob/main/go.mod#L26)
**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -342,3 +342,7 @@ replace github.com/go-chi/chi/v4 => github.com/go-chi/chi v4.0.0+incompatible
 
 // see https://github.com/ionos-cloud/sdk-go/issues/27
 exclude github.com/ionos-cloud/sdk-go/v6 v6.0.5851
+
+// This replace can be removed once it is patched to v0.3.2 upstream
+// see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c1a4f58263ec5d1c283827ce97286293aef75631/go.mod#L528
```
Comment (Aneurysm9) [inline go.mod:347]:
```suggestion
// see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/go.mod#L116
```

This is probably the better place to watch as it is the point at which the dependency enters the OTel realm, as a transitive dependency of one of the direct dependencies of the DataDog exporter.

**Review by Aneurysm9 — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (vasireddy99) [reply go.mod]: oh yes, even a better place to check. thank you

**Comment by vasireddy99** [comment]
Making this PR to draft.

File: go.mod [hunk 2]
```diff
@@ -343,3 +343,7 @@ replace github.com/go-chi/chi/v4 => github.com/go-chi/chi v4.0.0+incompatible
 
 // see https://github.com/ionos-cloud/sdk-go/issues/27
 exclude github.com/ionos-cloud/sdk-go/v6 v6.0.5851
+
+// This replace can be removed once it is patched to v0.3.2 upstream
+// see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/go.mod#L116
+replace github.com/theupdateframework/go-tuf => github.com/DataDog/go-tuf v0.3.0--fix-localmeta
```
Comment (Aneurysm9) [inline go.mod:349]:
```suggestion
replace github.com/theupdateframework/go-tuf v0.3.1 => github.com/DataDog/go-tuf v0.3.0--fix-localmeta
```

Can we specify the source version, too?  That way we should be in the clear when DataDog update to `>=v0.3.2` and can then remove this at our leisure.

**Pull request merged**